hw5/src/GrowList.java

```diff
@@ -0,0 +1,92 @@
+import java.util.*;
+//  GrowList is a mutable list that only gets longer.
+public class GrowList<E>{
+
+    private Map<Integer,E> values;
+
+    public Iterator values () {return new GrowListGen(this);}
+
+    public GrowList() { values = new HashMap<Integer,E>();}
+
+    // inner class
+    private static class GrowListGen<E> implements Iterator {
+
+    	private GrowList<E> growList;  // the GrowList being iterated
+        private int n;  // next term to consider
+    	
+    	GrowListGen(GrowList<E> gl){
+    		growList = gl;
+    		n = 0;
+    	}
+    	
+        public boolean hasNext() {
+            // TODO need to be fixed
+            // EFFECTS: Returns true if there are more elements to yield
+            //     else returns false
+            return true;
```

Comment:
I'm thinking this will be something like 

```
return n < growList.size();
```

I suppose this logic could also replace `growList.size() == 0 || growList.size() == n` in `next()`